lib/utils/validate.js

```diff
@@ -0,0 +1,43 @@
+const {promisify} = require("util");
```

Comment:
Could we rename this file to `fsHelper` or something alike?

In my opinion it doesn't really "validate" anything or throws errors if some validation failed. It currently just offers convenience wrappers for some fs APIs.